File: src/BikeApp.Api/BikeApp.Api/Model/CycleEvent.cs
using System;
using System.Collections.Generic;
using BikeApp.Api.Dto;

namespace BikeApp.Api.Model
{
	public class CycleEvent
	{
		public int Id { get; set; }
		public string Name { get; set; }
		public string Description { get; set; }
		public DateTime Date { get; set; }
		public string Location { get; set; }
		public List<int> Attendees { get; set; }
		public int MaxAttendees { get; set; }
	}
}

File: src/BikeApp.Client/src/app/models/cycle-event.ts
export interface CycleEvent {
  id: number;
  name: string;
  description: string;
  date: Date;
  location: string;
  attendees: number[]; // Array of User IDs
  maxAttendees: number;
}

export interface CycleEventAddRequest {
  name: string;
  description: string;
  date: Date;
  location: string;
  maxAttendees: number;
}

export interface CycleEventUpdateRequest {
  name?: string;
  description?: string;
  date?: Date;
  location?: string;
  maxAttendees?: number;
  attendees?: number[];
}